enterprise/api/v1/views.py

```diff
@@ -274,8 +273,20 @@ def enroll_learners_in_courses(self, request, pk):
         # Default subscription discount is 100%
         discount = serializer.validated_data.get('discount', 100.00)
 
-        # Get a list of user emails from the license info.
-        emails = {license_info['email'] for license_info in licenses_info}
+        enrollment_api_client = EnrollmentApiClient()
+
+        emails = set()
+        course_runs = {}
+        for index, info in enumerate(licenses_info):
+            emails.add(info['email'])
+            course_modes = [mode['slug'] for mode in enrollment_api_client.get_course_modes(info['course_run_key'])]
```

Comment:
Unfortunately as much as I want code reuse, this is going to heavily slow down your enrollment because you are now using the API client, and calling it individually for every single enrollment to do this. This is 100 API calls if you have 100 learners, even if you only have 1 course, to something you should be able to access from a database. I foresee this causing major problems.  Instead please do 2 things: (1) Add another loop before line 280. In this first loop, loop through your licenses and get all the courses. Use this to build your course_runs dictionary ahead of time and do NOT re-run the api call if you already have the course mode for a given course, since it doesn't vary.  (2) modify enrollment_api_client _in place_ to NOT make an HTTP call: have it directly make a database call. This will follow our general plan to incrementally rip out http calls where we should be using DB calls instead.

Reply:
yeah I had the same flag earlier, I just was not sure if we wanted to do it as part of this PR or we had a ticket, but we might as well take care of it now, edx-enterprise calling api to get enrollments is not great

Reply:
update 🤗